core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -231,9 +237,35 @@ void processSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateIn
     }
   }
 
+  private void onSubchannelHealth(HealthUtil.HealthCheckingListener hcListener,
+                                  HealthUtil.HealthStatus hcStatus) {
+    log.log(Level.FINE, "Received health status: " + hcStatus + " listener:" + hcListener);
+    Subchannel find = null;
+    for (SubchannelData subchannelData : subchannels.values()) {
+      if (subchannelData.healthCheckingListener == hcListener
+            && subchannelData.getState() == READY) {
+        find = subchannelData.subchannel;
+        break;
+      }
+    }
+    if (find == null) {
+      return;
+    }
+    if (hcStatus.servingStatus() != HealthUtil.ServingStatus.SERVING) {
```

Comment:
This is going to get triggered when the subchannel is created and everything is in UNKNOWN.

Reply:
Actually only READY status will be considered, otherwise find == null and return.